sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/BasicStatsEstimationSuite.scala

```diff
@@ -77,12 +92,21 @@ class BasicStatsEstimationSuite extends PlanTest with StatsEstimationTestBase {
                 max = Some(4),
                 nullCount = Some(0),
                 maxLen = Some(LongType.defaultSize),
-                avgLen = Some(LongType.defaultSize))))))
-    checkStats(range, expectedStatsCboOn = rangeStats, expectedStatsCboOff = rangeStats)
+                avgLen = Some(LongType.defaultSize),
+                histogram = histogram)))))
+    val extraConfig = Map(SQLConf.HISTOGRAM_ENABLED.key -> "true",
+      SQLConf.HISTOGRAM_NUM_BINS.key -> "3")
+    checkStats(range, expectedStatsCboOn = rangeStats,
+      expectedStatsCboOff = rangeStats, extraConfig)
   }
 
   test("range with negative step") {
     val range = Range(-10, -20, -2, None)
+    val histogramBins = new Array[HistogramBin](3)
+    histogramBins(0) = HistogramBin(-18.0, -16.0, 2)
+    histogramBins(1) = HistogramBin(-16.0, -12.0, 2)
+    histogramBins(2) = HistogramBin(-12.0, -10.0, 1)
```

Comment:
nit:
```
    val histogramBins = Array(
      HistogramBin(-18.0, -16.0, 2),
      HistogramBin(-16.0, -12.0, 2),
      HistogramBin(-12.0, -10.0, 1))
```

Reply:
Could you check if `range.numElements` and the total `ndv` are the same?

Reply:
Added assert to check if `range.numElements` and `ndv` are same